Automattic/wp-calypso
# Components: Move MaterialIcon to package

Related to https://github.com/Automattic/wp-calypso/issues/88182

## Proposed Changes

This moves the `MaterialIcon` component into `@automattic/components` to be consumed by other packages as well as Calypso. Part of migrating features-list to packages.

## Testing Instructions



- We handle everything in this PR, moving the component and updating the references
- There are several examples of use. Here are two with imgs below:

1. Go to `/home/[ site ]`, the first few icons under "Quick links":

<img width="500" alt="Screenshot 2024-03-11 at 1 00 06 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/af6cf784-dde8-45b3-a876-aa5428a2c273">


2. Go to `/me/purchases`, the items in the menu contain material-icon elements. The icons have a `display: none` currently, so they don't actually appear, but removing that will render as expected:

| hidden | visible |
|--------|--------|
| <img width="200" alt="Screenshot 2024-03-11 at 2 32 16 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/70c03810-36a6-4d48-8875-659c65848b7a"> | <img width="200" alt="Screenshot 2024-03-11 at 2 31 53 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/2fe84f49-4178-406a-9079-1e7b952f3afd"> | 

## Pre-merge Checklist
